### ISSUE TITLE
Audio gets stuck on seek backward

### ISSUE THREAD

**Issue by HaarigerHarald** [popcornmix/omxplayer#216]
I've noticed sometimes(every 3rd or 4th time) when i seek backward that the audio playback got stuck. This is the log:

> 13:46:31 T:18446744073222774972   DEBUG: Popped message member: SetPosition interface: org.mpris.MediaPlayer2.Player type: 1 path: /org/mpris/MediaPlayer2
> 13:46:31 T:18446744073222780592   DEBUG: OMXReader::SeekTime(4000) - seek ended up on time -2147483648
> 13:46:31 T:18446744073222781667   DEBUG: OMXClock::OMXStop
> 13:46:31 T:18446744073222786566   DEBUG: OMXClock::OMXSetSpeed(0.00) pause_resume:1
> 13:46:31 T:18446744073222793810    INFO: CDVDPlayerAudio::Decode dts:18877823 pts:18877823 size:536
> 13:46:31 T:18446744073222795094    INFO: CDVDPlayerAudio::Decode dts:18901043 pts:18901043 size:548
> 13:46:31 T:18446744073222807392   DEBUG: OMXClock::OMXMediaTime set config OMX_IndexConfigTimeCurrentAudioReference = 0.00
> 13:46:31 T:18446744073222808014   DEBUG: OMXThread::Run - Exited thread with  id -1305775008
> 13:46:31 T:18446744073222808459   DEBUG: OMXThread::StopThread - Thread stopped
> 13:46:31 T:18446744073222824537   DEBUG: COMXCoreComponent::Deinitialize : OMX.broadcom.video_scheduler handle 0xbd6028
> 13:46:31 T:18446744073222840990   DEBUG: COMXCoreComponent::Deinitialize : OMX.broadcom.video_decode handle 0x80de40
> 13:46:31 T:18446744073222843212   DEBUG: COMXCoreComponent::Deinitialize : OMX.broadcom.video_render handle 0x78b2a8
> 13:46:31 T:18446744073222843757   DEBUG: DllAvUtilBase: Using libavutil system library
> 13:46:31 T:18446744073222843917   DEBUG: DllAvCodec: Using libavcodec system library
> 13:46:31 T:18446744073222844004   DEBUG: DllAvFormat: Using libavformat system library
> 13:46:31 T:18446744073222847319   DEBUG: COMXCoreComponent::Initialize OMX.broadcom.video_decode input port 130 output port 131 m_handle 0x5080c0
> 13:46:31 T:18446744073222850774   DEBUG: COMXCoreComponent::AllocInputBuffers component(OMX.broadcom.video_decode) - port(130), nBufferCountMin(1), nBufferCountActual(60), nBufferSize(81920), nBufferAlignmen(16)
> 13:46:31 T:18446744073222870653   DEBUG: COMXVideo::Open - decoder_component(0x0x5080c0), input_port(0x82), output_port(0x83) deinterlace 1 hdmiclocksync 0
> 13:46:31 T:18446744073222871060   DEBUG: OMXThread::Create - Thread with id -1305775008 started
> 13:46:31 T:18446744073222871528   DEBUG: Seeked 4000014 4000000 0
> 13:46:31 T:18446744073222871936   DEBUG: Normal M:0 (A:-4503599627370496 V:-4503599627370496) P:1 A:0.00 V:0.00/T:0.20 (0,0,0,0) A:0% V:0% (0.00,6.34)
> 13:46:31 T:18446744073222872152   DEBUG: COMXPlayer::HandleMessages - player started RESET
> 13:46:31 T:18446744073222872471   DEBUG: OMXClock::OMXReset audio / video : 1 / 1 wait mask 0->3 state : 2->1
> 13:46:31 T:18446744073223256736    INFO: CDVDPlayerVideo::Decode dts:5005000 pts:5005000 cur:5005000, size:40800
> 13:46:31 T:18446744073223257118   DEBUG: OMXVideo::Decode VDec : setStartTime 5.005000
> 13:46:31 T:18446744073223259557   DEBUG: Normal M:0 (A:-4503599627370496 V:5005000) P:1 A:0.00 V:5.01/T:0.20 (0,0,0,1) A:0% V:0% (0.00,6.34)
> 13:46:31 T:18446744073223264061    INFO: CDVDPlayerVideo::Decode dts:5046708 pts:5046708 cur:5046708, size:5277
> 13:46:31 T:18446744073223271342    INFO: CDVDPlayerAudio::Decode dts:5038730 pts:5038730 size:568
> 13:46:31 T:18446744073223273131    INFO: CDVDPlayerAudio::Decode dts:5061950 pts:5061950 size:554
> 13:46:31 T:18446744073223275753    INFO: CDVDPlayerAudio::Decode dts:5085170 pts:5085170 size:549
> 13:46:31 T:18446744073223278578    INFO: CDVDPlayerAudio::Decode dts:5108390 pts:5108390 size:579
> 13:46:31 T:18446744073223281200   DEBUG: Normal M:0 (A:5108390 V:5046708) P:1 A:5.11 V:5.05/T:0.20 (0,0,1,1) A:0% V:0% (0.00,6.34)
> 13:46:31 T:18446744073223281605   DEBUG: Resume 5.11,5.05 (0,0,1,1) EOF:0 PKT:(nil)
> 13:46:31 T:18446744073223281794   DEBUG: OMXClock::OMXSetSpeed(1.00) pause_resume:1
> 13:46:31 T:18446744073223284091   DEBUG: COMXAudio::Decode ADec : setStartTime 18.831385
> 13:46:31 T:18446744073223290762    INFO: COMXAudio::ApplyVolume - Volume=1.00 (\* 1.00 \* 1.00)
> 13:46:31 T:18446744073223291128    INFO: CDVDPlayerAudio::Decode dts:5131610 pts:5131610 size:571

notice: "DEBUG: Seeked 4000014 4000000 0" and "COMXAudio::Decode ADec : setStartTime 18.831385"

It seems to me that the Audiodecode Thread somhow got locked on seeking and when it gets unlocked the local variables pts and dts have a wrong value in them. What I don't understand is why this only happens when I seek backward.

Long story short, this fixed it for me but I'm not sure if this is a good solution:

OMXPlayerAudio.cpp:

```
while((int) m_decoder->GetSpace() < decoded_size)
{
    OMXClock::OMXSleep(10);
        if(m_flush_requested) return true;
}

+if(pkt->dts<dts || pkt->pts<pts)
+{
+   return true;
+}

int ret = 0;
ret = m_decoder->AddPackets(decoded, decoded_size, dts, pts, m_pAudioCodec->GetFrameSize());
```
